mindmeld/active_learning/classifiers.py

```diff
@@ -531,3 +680,122 @@ def _update_eval_stats_intent_level(
                         self.class_level_statistic
                     ][i]
                 }
+
+    def entity_recognizers_fit_eval(
+        self,
+        sampled_queries: ProcessedQueryList,
+        unsampled_queries: ProcessedQueryList,
+        test_queries: ProcessedQueryList,
+        domain_to_intents: Dict,
+        domain_to_intent2id: Dict,
+        entity2id: Dict,
+        heuristic: Heuristic,
+    ):
+        """Fit and evaluate the entity recognizer.
+        Args:
+            sampled_queries (ProcessedQueryList): List of Sampled Queries.
+            unsampled_queries (ProcessedQueryList): List of Unsampled Queries.
+            test_queries (ProcessedQueryList): List of Test Queries.
+            domain_to_intents (Dict): Dictionary mapping domain to list of intents.
+            domain_to_intent2id (Dict): Dictionary mapping intents to IDs.
+            entity2id (Dict): Dictionary mapping entities to IDs.
+
+        Returns:
+            ic_queries_prob_vectors (List[List]): List of probability distributions
+                for unsampled queries.
+            ic_eval_test_dict (Dict): Dictionary mapping a domain (str) to the
+                associated ic_eval_test object.
+        """
+        er_eval_test_dict = {}
+        unsampled_idx_preds_pairs = {}
+        for domain, intents in domain_to_intents.items():
+            for intent in intents:
+                # Filter Queries
+                (
+                    _,
+                    filtered_sampled_queries,
+                ) = DataBucket.filter_queries_by_nlp_component(
+                    query_list=sampled_queries,
+                    component_type="intent",
+                    component_name=intent,
+                )
+                (
+                    filtered_unsampled_queries_indices,
+                    filtered_unsampled_queries,
+                ) = DataBucket.filter_queries_by_nlp_component(
+                    query_list=unsampled_queries,
+                    component_type="intent",
+                    component_name=intent,
+                )
+                _, filtered_test_queries = DataBucket.filter_queries_by_nlp_component(
+                    query_list=test_queries,
+                    component_type="intent",
+                    component_name=intent,
+                )
+                # Train
+                er = self.nlp.domains[domain].intents[intent].entity_recognizer
+                try:
+                    er.fit(queries=filtered_sampled_queries)
+                except:
+                    pass
+                # Evaluate Test Queries
+                er_eval_test = er.evaluate(queries=filtered_test_queries)
+                er_eval_test_dict[f"{domain}.{intent}"] = er_eval_test
+
+                # Get Probability Vectors
```

Comment:
Why are we setting probability vectors for unsampled queries here?

Reply:
We're evaluating on unsampled queries given the model trained on the sampled queries so far. These probabilities/confidences are used for active learning selection